### PR TITLE
[REF] account: Making the creation of Journal Items for CABA inheritable.

### PR DESCRIPTION
**Pull request by hbto**
# Main

Inheriting the creation of Cash Basis (CABA) Journal Entries is hard to be done and the current way of doing it can lead to the creation of a huge number of lines that are not quite useful and its creations can lead to a lengthy process of creation of payment when several invoices with a huge amount of invoice lines involved or to a lengthy process of cancelation of payment because of reversion of the CABA, in case of l10n_mx, many users have opted for using l10n_mx_edi_avoid_reversal_entry, and then this causes a lengthy process of deletions of CABA entries.

# Justification

CABA Entry is divided into two parts when it is created
- Tax Part: The part that deals with the taxes themselves (this PR does not mess with them).
- Base Part: The part of the base of the taxes (what this PR is all about)

Currently Base Part is creating a big deal of line and we want to be able on our side to change this behavior.

By providing a better inheritability on the `create_tax_cash_basis_entry` method Developers can leverage on it and make their own implementation according to their local needs.

**This PR does not change the current behavior of `create_tax_cash_basis_entry` method**

# Enterprise-MX MR

https://git.vauxoo.com/vauxoo/enterprise-mx/merge_requests/703

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by hbto** [comment]
Hello @oco-odoo & @nim-odoo this PR supersedes https://github.com/odoo/odoo/pull/44444 as requested at https://github.com/odoo/odoo/pull/44444#issuecomment-590794543
Requested changes https://github.com/odoo/odoo/pull/44444#pullrequestreview-358073739 in original PR have been applied

File: addons/account/models/account_move.py [hunk 1]
```diff
@@ -1843,34 +1881,20 @@ def create_tax_cash_basis_entry(self, percentage_before_rec):
...
-                                    'credit': rounded_amt > 0 and rounded_amt or 0.0,
-                                    'debit': rounded_amt < 0 and abs(rounded_amt) or 0.0,
-                                    'account_id': account_id.id,
-                                    'tax_exigible': True,
-                                    'move_id': newly_created_move.id,
-                                    'currency_id': line.currency_id.id,
-                                    'amount_currency': self.amount_currency and line.currency_id.round(-line.amount_currency * amount / line.balance) or 0.0,
-                                    'partner_id': line.partner_id.id,
-                                })
+                            for tax in line.tax_ids.filtered(lambda t: t.tax_exigibility == 'on_payment'):
```
Comment (nim-odoo) [inline addons/account/models/account_move.py]: I would not filter, so we keep the exact same behavior.

Comment (hbto) [reply addons/account/models/account_move.py]: @nim-odoo I suggested it in the previous PR 

@oco-odoo agreed on it at comment https://github.com/odoo/odoo/pull/44444#pullrequestreview-362028861

As per commit https://github.com/odoo/odoo/commit/6424111f is supporting it on enterprise.

Comment (hbto) [reply addons/account/models/account_move.py]: @nim-odoo in any case to comply with your request I have made the fix,

and in a later PR if you agree we can push that proposal.

Regards

**Comment by hbto** [comment]
@nim-odoo

File: addons/account/models/account_move.py [hunk 2]
```diff
@@ -1843,34 +1881,20 @@ def create_tax_cash_basis_entry(self, percentage_before_rec):
...
-                                })
+                                # We want to group base lines as much as
+                                # possible to avoid creating too many of them.
+                                # This will result in a more readable report
+                                # tax and less cumbersome to analyse.
+                                key = self._get_tax_cash_basis_base_key(tax, move, line)
+                                caba_amount_dict[key] += rounded_amt
+                                caba_amount_currency_dict[key] += line.currency_id.round(line.amount_currency * amount / line.balance) if line.currency_id and self.amount_currency else 0.0
+
+            if caba_amount_dict:
```
[the remark this answers is not in the document]
Comment (hbto) [reply addons/account/models/account_move.py]: This doesn't make sense @jpp-odoo

**Comment by hbto** [comment]
@jpp-odoo @nim-odoo 

Changes were made

Request on comment https://github.com/odoo/odoo/pull/46390#discussion_r386978537 doesn't make sense.

Regards

File: addons/account/models/account_move.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jpp-odoo) [reply addons/account/models/account_move.py]: Indeed sorry, a bad copy/paste.

**Comment by hbto** [comment]
@jpp-odoo & @nim-odoo conflicts solved

**Pull request closed without merging**